### PR TITLE
Added lifecycle/activation support to opentracing-jdbc

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Motivation and Context

opentracing-jdbc is now official a level-4 instrumentation library, according to #353 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: opentracing-jdbc/opentracing-jdbc/src/main/java/org/zalando/opentracing/jdbc/span/SpanDecorator.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import io.opentracing.Span;
 import org.apiguardian.api.API;
 
+import java.sql.ResultSet;
```
Comment (whiskeysierra) [inline opentracing-jdbc/opentracing-jdbc/src/main/java/org/zalando/opentracing/jdbc/span/SpanDecorator.java]: Codacy Issue found: [Avoid unused imports such as 'java.sql.ResultSet'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4673604)

File: opentracing-jdbc/opentracing-jdbc/src/test/java/org/zalando/opentracing/jdbc/DefaultActivationTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import org.zalando.opentracing.proxy.ProxyTracer;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
```
Comment (whiskeysierra) [inline opentracing-jdbc/opentracing-jdbc/src/test/java/org/zalando/opentracing/jdbc/DefaultActivationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers.hasEntry'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4673604)

File: opentracing-jdbc/opentracing-jdbc/src/test/java/org/zalando/opentracing/jdbc/DefaultActivationTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
```
Comment (whiskeysierra) [inline opentracing-jdbc/opentracing-jdbc/src/test/java/org/zalando/opentracing/jdbc/DefaultActivationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers.not'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4673604)

File: opentracing-jdbc/opentracing-jdbc/src/test/java/org/zalando/opentracing/jdbc/DefaultActivationTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
```
Comment (whiskeysierra) [inline opentracing-jdbc/opentracing-jdbc/src/test/java/org/zalando/opentracing/jdbc/DefaultActivationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers.is'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4673604)

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**